akka-projection-core-test/src/test/scala/akka/projection/internal/metrics/tools/InternalProjectionStateMetricsSpec.scala

```diff
@@ -134,25 +134,32 @@ object InternalProjectionStateMetricsSpec {
     val adaptedHandlerStrategy: HandlerStrategy = offsetStrategy match {
       case ExactlyOnce(_) =>
         handlerStrategy match {
-          case SingleHandlerStrategy(handlerFactory) => {
+          case singleHandlerStrategy: SingleHandlerStrategy[Envelope] @unchecked => {
             val adaptedHandler = () =>
               new Handler[Envelope] {
-                override def process(envelope: Envelope): Future[Done] = handlerFactory().process(envelope).flatMap {
-                  _ =>
+                override def process(envelope: Envelope): Future[Done] =
+                  singleHandlerStrategy.handlerFactory().process(envelope).flatMap { _ =>
                     offsetStore.saveOffset(projectionId, envelope.offset)
-                }
+                  }
               }
             SingleHandlerStrategy(adaptedHandler)
           }
-          case GroupedHandlerStrategy(handlerFactory, afterEnvelopes, orAfterDuration) => {
+          case groupedHandlerStrategy: GroupedHandlerStrategy[Envelope] @unchecked => {
             val adaptedHandler = () =>
               new Handler[immutable.Seq[Envelope]] {
                 override def process(envelopes: immutable.Seq[Envelope]): Future[Done] =
-                  handlerFactory().process(envelopes).flatMap { _ =>
-                    offsetStore.saveOffset(projectionId, envelopes.last.offset)
-                  }
+                  groupedHandlerStrategy
+                    .handlerFactory()
+                    .asInstanceOf[Handler[Seq[Envelope]]]
```

Comment:
shouldn't that be `immutable.Seq`? needed for Scala 2.12

Reply:
btw, isn't it strange that this cast is needed? The `groupedHandlerStrategy` already has the `Envelope` type

Reply:
Dang, good catch. This was a leftover with my battle with `@unchecked` which I did not revert.